### PR TITLE
Uglify 2 to 3 - READY FOR TESTING / MERGE - Scheduled for merge BUT REMEMBER package-json.lock

### PR DESCRIPTION
**Pull request by JohnMcLear**
Fixes a security dep issue.

Discussion @ https://github.com/ether/etherpad-lite/pull/3808

Uglify supports buffers now so could remove some cruft.

Ran tests.

### CODE REVIEW

**Comment by muxator** [comment]
What's the difference between this one (3809) and the previous one (3808)?

- https://github.com/ether/etherpad-lite/pull/3809/files <-- this one
- https://github.com/ether/etherpad-lite/pull/3808/files <-- the previous one. Same modifications

**Comment by JohnMcLear** [comment]
This one should have had package-lock.json but I'm an idiot :D

**Comment by muxator** [comment]
Due to merge conflicts, integrated manually with 1f5fd6e500420b13223de0a75c48acacb413f92c.

Closing this.

File: src/node/utils/Minify.js [hunk 1]
```diff
@@ -406,10 +406,7 @@ function getFile(filename, callback) {
 
 function compressJS(content)
 {
-  var decoder = new StringDecoder('utf8');
-  var code = decoder.write(content); // convert from buffer to string
-  var codeMinified = uglifyJS.minify(code, {fromString: true}).code;
-  return codeMinified;
+  return UglifyJS.minify(content);
```
Comment (muxator) [inline src/node/utils/Minify.js:409]: This is wrong.

On top of the file uglifyjs is imported with a lowercase letter.
```js
var uglifyJS = require("uglify-js");
```

How did it work?

I will force push on develop.

**Comment by muxator** [comment]
John, this did not work.

```
[2020-04-03 02:29:58.458] [ERROR] console - TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be one of type string or Buffer. Received type object
    at write_ (_http_outgoing.js:618:11)
    at ServerResponse.write (_http_outgoing.js:586:15)
    at /mnt/md/hot-data/documenti/codice/etherpad/etherpad-lite/src/node/utils/Minify.js:236:15
    at /mnt/md/hot-data/documenti/codice/etherpad/etherpad-lite/src/node/utils/Minify.js:388:7
    at getFile (/mnt/md/hot-data/documenti/codice/etherpad/etherpad-lite/src/node/utils/Minify.js:401:5)
    at getFileCompressed (/mnt/md/hot-data/documenti/codice/etherpad/etherpad-lite/src/node/utils/Minify.js:379:3)
    at /mnt/md/hot-data/documenti/codice/etherpad/etherpad-lite/src/node/utils/Minify.js:229:9
    at statFile (/mnt/md/hot-data/documenti/codice/etherpad/etherpad-lite/src/node/utils/Minify.js:308:5)
    at minify (/mnt/md/hot-data/documenti/codice/etherpad/etherpad-lite/src/node/utils/Minify.js:201:3)
    at Layer.handle [as handle_request] (/mnt/md/hot-data/documenti/codice/etherpad/etherpad-lite/src/node_modules/express/lib/router/layer.js:95:5)
```

I will kill the change. It never happened.

Please do a PR **from a clean clone** to also validate the installation.

**Comment by muxator** [comment]
I reopened for mistake.

This has to be done again from a clean clone based on the latest version.

**Pull request closed without merging**